File: src/main/scala/mffs/field/module/ItemModuleSponge.java
package mffs.field.module;

import resonant.api.mffs.IProjector;
import mffs.base.ItemModule;
import net.minecraft.block.Block;
import net.minecraft.block.BlockFluid;
import net.minecraft.item.ItemStack;
import net.minecraft.tileentity.TileEntity;
import net.minecraft.world.World;
import net.minecraftforge.fluids.BlockFluidBase;
import universalelectricity.core.transform.vector.Vector3;

import java.util.Set;

public class ItemModuleSponge extends ItemModule
{
	public ItemModuleSponge(int i)
	{
		super(i, "moduleSponge");
		this.setMaxStackSize(1);
	}

	@Override
	public boolean onProject(IProjector projector, Set<Vector3> fields)
	{
		if (projector.getTicks() % 60 == 0)
		{
			World world = ((TileEntity) projector).worldObj;

			if (!world.isRemote)
			{
				for (Vector3 point : projector.getInteriorPoints())
				{
					Block block = Block.blocksList[point.getBlockID(world)];

					if (block instanceof BlockFluid || block instanceof BlockFluidBase)
					{
						point.setBlock(world, 0);
					}
				}
			}
		}

		return super.onProject(projector, fields);
	}

	@Override
	public boolean requireTicks(ItemStack moduleStack)
	{
		return true;
	}
}

File: src/main/scala/mffs/field/module/ItemModuleSponge.scala
package mffs.field.module

import java.util

import mffs.api.machine.Projector
import mffs.base.ItemModule
import mffs.field.BlockProjector
import nova.core.fluid.FluidBlock
import nova.core.network.NetworkTarget.Side
import nova.core.util.transform.Vector3i

import scala.collection.convert.wrapAll._

class ItemModuleSponge extends ItemModule {
	setMaxCount(1)

	override def getID: String = "moduleSponge"

	override def onCreateField(projector: Projector, field: util.Set[Vector3i]): Boolean = {
		val proj = projector.asInstanceOf[BlockProjector]

		if (proj.getTicks % 60 == 0) {
			val world = proj.world

			if (Side.get().isServer) {
				for (point <- projector.getInteriorPoints) {
					val block = world.getBlock(point).get

					if (block.isInstanceOf[FluidBlock]) {
						world.removeBlock(point)
					}
				}
			}
		}

		return false
	}

	override def requireTicks = true
}